src/sage/modular/drinfeld_modform/tutorial.py

```diff
@@ -0,0 +1,228 @@
+
+    \phi^w : T \mapsto T + g_1(w)\tau + \cdots
+    + g_{r - 1}(w)\tau^{r-1} + g_{r}(w)\tau^{r}.
+
+This Drinfeld module is called the *universal
+Drinfeld `\mathbb{F}_q[T]`-module over `\Omega^r(\mathbb{C}_{\infty})`*
+and its coefficients
+`g_i : \Omega^r(\mathbb{C}_{\infty}) \rightarrow \mathbb{C}_{\infty}`
+are rigid analytic functions satisfying the invariance property:
+
+.. MATH::
+
+    g_i(\gamma(w)) = j(\gamma, w)^{1 - q^i} g_i(w),
+    ~\forall \gamma\in \mathrm{GL}_r(A)
+
+where `g_{r}(w)` never vanishes. Moreover, these coefficients `g_i`
+admits an expansion at infinity, analogous to `q`-expansion principle
+for classical modular forms. The functions `g_i` are known as the
+*coefficients forms at* `T`. More generally, the coefficients of the
+image `\phi^w_a` for any `a\in A` are called the *coefficient forms at*
+`a` and they are an algebraic combination of the coefficient forms at
+`T`.
+
+In the rank two case, the expansion at infinity is of the form
+
+.. MATH::
+
+    g_i(w) = \sum_{i = 0}^{\infty} a_n(g_i)u(w)^i
+
+where `u(w) := e(w)^{-1}` and `e` is the exponential of the Carlitz
+module `\rho:T\mapsto T + \tau`. The analytic parameter `u` is called
+the *parameter at infinity*.
+
+A *Drinfeld modular form* of rank `r`, weight `k`, type `m` for
+`\mathrm{GL}_r(A)` is a rigid analytic function
+`f:\Omega^r(\mathbb{C}_{\infty}) \rightarrow \mathbb{C}_{\infty}`
+such that
+
+* `f(\gamma(w)) = \mathrm{det}(\gamma)^m j(\gamma, w)^k f(w)` for all
+  `\gamma` in `\mathrm{GL}_r(A)` and
+  `w\in \Omega^r(\mathbb{C}_{\infty})`;
+
+* `f` is holomorphic at infinity.
+
+The second condition is similar to the classical case. In the rank two
+situation, this expansion is simply given by
+`f = \sum_{n\geq 0} a_n(f) u^n` where `a_n(f)\in \mathbb{C}_{\infty}`.
+
+The reader is refered to
+part I of [BRP2018]_ for more information about the analytic theory of
+Drinfeld modular form of arbitrary rank.
+
+.. RUBRIC:: Ring of Drinfeld modular forms
+
+Letting `M_k^{r, m}(\mathrm{GL}_r(A))` denote the space of rank `r`,
+weight `k\in (q - 1)\mathbb{Z}` and type `m~(\mathrm{mod}~q-1)`
+Drinfeld modular forms, we define
+
+.. MATH::
+
+    M^{r, 0}(\mathrm{GL}_r(A)) :=
+    \bigoplus_{k\in ZZ} M_k^{r}(\mathrm{GL}_r(A))
+
+to be the graded ring of all Drinfeld modular forms of type 0. The
+graduation is given by the weight of a modular form. Similarly, we let
+`M^{r}(\mathrm{GL}_r(A)) \supset M^{r, 0}(\mathrm{GL}_r(A))` be the ring
+of all Drinfeld modular forms of arbitrary type. By
+theorem 17.5 in part III of [BRP2018]_, we have
+
+.. MATH::
+
+    M^{r, 0}(\mathrm{GL}_r(A))
+    = \mathbb{C}_{\infty}[g_1,\ldots, g_{r-1}, g_{r}].
+
+and
+
+.. MATH::
+
+    M^{r, 0}(\mathrm{GL}_r(A))
+    = \mathbb{C}_{\infty}[g_1,\ldots, g_{r-1}, h_{r}].
+
+where `h_r` is a weight `(q^r - 1)/(q - 1)` modular forms of type `1`
+which is a `(q-1)`-root of `g_r`.
+
+To create the ring of type zero forms in SageMath we use the class
+`DrinfeldModularForms`::
```

Comment:
```suggestion
To create the ring of type zero and rank `r` Drinfeld modular forms in
SageMath, one uses the class `DrinfeldModularForms`::
```

Also, shouldn't you write class:`.../DrinfeldModularForms`?